Client/Frontend/Browser/BrowserLocationView.swift

```diff
@@ -267,8 +274,25 @@ class BrowserLocationView : UIView, ToolbarTextFieldDelegate {
         return true
     }
 
-    func textFieldDidLongPress(textField: ToolbarTextField) {
-        delegate?.browserLocationViewDidLongPressLocation(self)
+
+    func longPress(gestureRecognizer: UIGestureRecognizer) {
+        if gestureRecognizer.state == .Began {
+            delegate?.browserLocationViewDidLongPressLocation(self, completion: { success in
```

Comment:
The problem we have here is that the order in which `longPress:` and `textFieldShouldBeginEditing` are called can change. Sometimes `textFieldShouldBeginEditing` will get called before we have started handling the longPress gesture, and sometimes the long press gesture ends before `textFieldShouldBeginEditing` is called, one resulting in the text field becoming editable when the action menu is displayed, the other resulting in only the text field becoming editable and no action menu being displayed. 

we need `shouldRecognizeSimultaneouslyWithGestureRecognizer` as otherwise the UILongPressGestureRecognizer that be default exists on the text field (for copy/paste etc) will be the only gesture recognizer that gets called and our one won't, ensuring that the long press functionality will never get called

Because we never get notified when the default UITextFieldDelegate gesture recognizers are fired, we have no way of cancelling them if we have detected OUR gesture recognizer, which means the only method we have of preventing editing is to know that we have a gesture we need to handle.

The only way I found to handle this was to make a note when we realise we need to handle the gesture, use that to determine if we should start editing the text field and only dismiss the state once we know for sure that all the required things have been finished, which made me want to know when we were done with the action view controller.

I guess we could change that to change the long press status flag once we have handled the `textFieldShouldBeginEditing`, but then there is no way of know for sure that the gesture recognizer has been called yet.

Other solution would be to return to having 2 views on this, one clear `UIView` that sits over the top of the `UITextField` that handles the long press and tap gesture and explicitly calls `firstResponder` on the text field and hides itself (and remembers to reshow itself when the text field dismisses first responder. Maybe a better approach state wise, What do you think?